components/list/list-item-drag-mixin.js

```diff
@@ -33,86 +95,299 @@ export const ListItemDragMixin = superclass => class extends superclass {
 		}
 	}
 
-	moveAfter(itemKey) {
+	moveItemAfterMe(itemKey) {
 		if (itemKey === undefined) {
 			return;
 		}
-		this._dispatchDragEvent(itemKey);
+		this._moveItem(itemKey, this.key, false);
 	}
 
-	moveBefore(itemKey) {
+	moveItemBeforeMe(itemKey) {
 		if (itemKey === undefined) {
 			return;
 		}
 
-		this._dispatchDragEvent(itemKey);
+		this._moveItem(itemKey, this.key, true);
+	}
+
+	_dragAreaClick(e) {
+		this.shadowRoot.querySelector(`#${this._itemDragId}`).activateKeyboardMode();
+		e.preventDefault();
 	}
 
-	// TODO
-	_copy() {
+	_dragBottomEnter(e) {
+		e.dataTransfer.dropEffect = 'move';
+		const dropSpots = dropSpotsFactory();
+		dropSpots.setDestination(this, move.below);
+		this._inBottomArea = true;
+	}
 
+	_dragEnd(e) {
+		const dropSpot = dropSpotsBlowUp();
+		this.beingDragged = false;
+		this.tilt = false;
+		if (dropSpot.shouldDrop(e.timeStamp)) {
+			this._moveItem(dropSpot.targetKey, dropSpot.destinationKey, dropSpot.moveBeforeDestination);
+		}
 	}
 
-	_dispatchDragEvent(detail) {
+	_dragHandleActions(e) {
+		let destinationKey;
+		switch (e.detail.action) {
+			case dragActions.active:
+				this._keyboardMode = true;
+				break;
+			case dragActions.cancel:
+			case dragActions.save:
+				this._keyboardMode = false;
+				break;
+			case dragActions.up:
+				destinationKey = this.previousElementSibling && this.previousElementSibling.key;
+				this._moveItem(this.key, destinationKey, true);
+				break;
+			case dragActions.down:
+				destinationKey = this.nextElementSibling && this.nextElementSibling.key;
+				this._moveItem(this.key, destinationKey);
+				break;
+			case dragActions.first:
+				while (this.previousElementSibling) {
+					destinationKey = this.previousElementSibling;
+				}
+				this._moveItem(this.key, destinationKey, true);
+				break;
+			case dragActions.last:
+				while (this.nextElementSibling) {
+					destinationKey = this.nextElementSibling;
+				}
+				destinationKey = destinationKey && destinationKey.key;
+				this._moveItem(this.key, destinationKey);
+				break;
+			default:
+				break;
+		}
+
+	}
+
+	_dragLowerEnter(e) {
+		e.dataTransfer.dropEffect = 'move';
+		if (this._inBottomArea) {
+			const dropSpots = dropSpotsFactory();
+			dropSpots.setDestination(this, move.above);
+			this._inBottomArea = false;
+		}
+	}
+
+	_dragOver(e) {
+		const dropSpots = dropSpotsFactory();
+		dropSpots.updateTime(e.timeStamp);
+		e.preventDefault();
+	}
+
+	_dragStart(e) {
+		e.dataTransfer.setData('text/plain', `${this.dragText}`);
+		e.dataTransfer.effectAllowed = 'move';
+
+		const nodeImage = this.shadowRoot.querySelector('.d2l-list-item-drag-image') || this;
```

Comment:
This is where we define the image. If the list item defines a specific image use that. Otherwise just use the entire element.